### PR TITLE
Update CMakeLists.txt

### PR DESCRIPTION
**Pull request by Thunder07**
target_include_directories allows include propagation in the rest of the project

### CODE REVIEW

File: build_cmake/FrameworkOpenAl/CMakeLists.txt [hunk 1]
```diff
@@ -19,4 +19,5 @@ set(SRC_FILES
 )
 
 add_library(Framework_OpenAl ${SRC_FILES})
+target_link_libraries(Framework_OpenAl PUBLIC ${OPENAL_LIBRARY})
```
Comment (jpd002) [inline build_cmake/FrameworkOpenAl/CMakeLists.txt]: Could that be moved in a separate commit? It doesn't really have anything to do with making sure `string_cast_win1252.cpp` compiles.

Comment (Thunder07) [reply build_cmake/FrameworkOpenAl/CMakeLists.txt]: moved to the previous commit

**Comment by jpd002** [comment]
Thanks for this, the `target_include_*` commands are great!

**Comment by Thunder07** [comment]
yup, they're great, you can even create an `interface` library that contain only headers and dependencies to link against :)
